### PR TITLE
Maple Upgrade

### PR DESCRIPTION
**Pull request by BbrSofiane**
- [x] In the plugin’s `setup.py`, bump the “tutor” version that the plugin depends on.
- [x] In the `__about__.py` module, bump the version number to the next major release
- [x] Replace all instances of the previous release name (“lilac”) with the new one (“maple”) in the entire repo: `git grep -i lilac`.
- [x] Nginx-specific patches should be deleted, as we no longer run Nginx in Maple (see [PR 2](https://github.com/overhangio/tutor/pull/508/))
- [ ] Containers should run in unprivileged mode on Kubernetes (see [PR 1](https://github.com/overhangio/tutor/pull/505/files))
- [x] Upgrade server and client images
- [x] `MINIO_ACCESS_KEY` and `MINIO_SECRET_KEY` are deprecated.  Please use `MINIO_ROOT_USER` and `MINIO_ROOT_PASSWORD`

# Tests
- [x] Dev machine using tutor local
- [x] Dev machine using tutor dev
- [x] Dev machine using tutor k8s
- [ ] Remote server using tutor local and HTTPS
- [ ] Remote server using tutor local, HTTPS and S3 gateway

### CODE REVIEW

**Comment by BbrSofiane** [comment]
[Embedded MinIO Browser removed replaced with MinIO Console project](https://github.com/minio/minio/releases/tag/RELEASE.2021-07-08T01-15-01Z)

File: tutorminio/patches/local-docker-compose-dev-services [hunk 1]
```diff
@@ -1,6 +1,6 @@
 minio:
   ports:
-    - "127.0.0.1:9000:9000"
+    - "127.0.0.1:9001:9001"
```
Comment (regisb) [inline tutorminio/patches/local-docker-compose-dev-services:3]: :+1: could you amend the README accordingly?

File: README.rst [hunk 2]
```diff
@@ -24,8 +24,8 @@ Configuration
 - ``MINIO_BUCKET_NAME`` (default: ``"openedx"``)
 - ``MINIO_FILE_UPLOAD_BUCKET_NAME`` (default: ``"openedxuploads"``)
 - ``MINIO_HOST`` (default: ``"minio.{{ LMS_HOST }}"``)
-- ``MINIO_DOCKER_IMAGE`` (default: ``"docker.io/minio/minio:RELEASE.2021-06-07T21-40-51Z"``)
-- ``MINIO_MC_DOCKER_IMAGE`` (default: ``"docker.io/minio/mc:RELEASE.2021-06-08T01-29-37Z"``)
+- ``MINIO_DOCKER_IMAGE`` (default: ``"docker.io/minio/minio:RELEASE.2021-10-23T03-28-24Z"``)
+- ``MINIO_MC_DOCKER_IMAGE`` (default: ``"docker.io/minio/mc:RELEASE.2021-10-07T04-19-58Z"``)
```
Comment (regisb) [inline README.rst:28]: I see that there are even more recent tags:
https://hub.docker.com/r/minio/mc/tags
https://hub.docker.com/r/minio/minio/tags

Could you try running these versions?

Comment (BbrSofiane) [reply README.rst]: Done

File: tutorminio/patches/local-docker-compose-dev-services (hunk 1 above)
Comment (BbrSofiane) [reply tutorminio/patches/local-docker-compose-dev-services:3]: Done

File: tutorminio/patches/caddyfile [hunk 3]
```diff
@@ -1,4 +1,4 @@
 # MinIO
-{{ MINIO_HOST }}{% if not ENABLE_HTTPS %}:80{% endif %} {
-    reverse_proxy nginx:80
+{{ MINIO_HOST }}{$default_site_port} {
+    import proxy "minio:9001"
```
Comment (regisb) [inline tutorminio/patches/caddyfile:3]: Wait... What is the difference between the server and the console ports? If I understand correctly, the console port is for accessing the web UI, right? The problem is that other services, such as the LMS/CMS, need to be able to access the server from the https://minio.mylms.com address. So the MINIO_HOST should point to the minio server, and not the console.

What this means is that we might need to expose the MinIO console at a different web host. Something like:

```
{{ MINIO_CONSOLE_HOST }}{$default_site_port} {
    import proxy "minio:9000"
{{ MINIO_CONSOLE_HOST }}{$default_site_port} {
    import proxy "minio:9001"
}
```

Do you understand my point?

File: tutorminio/patches/local-docker-compose-dev-services (hunk 1 above)
Comment (regisb) [reply tutorminio/patches/local-docker-compose-dev-services:3]: See my comment above: we might have to expose both the 9000 and the 9001 ports.

**Review by regisb — CHANGES_REQUESTED** [review]
I tested this PR and failed to upload media files. I can attempt to make changes to make it work. It probably means running the MinIO console at a different domain name.

**Comment by regisb** [comment]
Closed by https://github.com/overhangio/tutor-minio/pull/12

**Pull request closed without merging**